src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -489,6 +510,17 @@ exprt gdb_value_extractort::get_pointer_value(
       return result_expr;
     }
 
+    // pointer to function
+    if(expr.type().subtype().id() == ID_code)
+    {
+      const auto target_expr =
+        get_pointer_to_function_value(expr, value, location);
+      CHECK_RETURN(target_expr.id() != ID_nil);
+      const auto result_expr = address_of_exprt(target_expr);
+      CHECK_RETURN(result_expr.type() == zero_expr.type());
+      return result_expr;
```

Comment:
This needs to be `return std::move(result_expr);` if you use `address_of_exprt` above, or can stay as-is if using `exprt result_expr = ...`. I'd say this is a case where `auto` above really hurts you, because it hides a lot of this.